From 51dc7876026d012e053121f4c6419f9568c05d14 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Tue, 21 Oct 2025 15:33:47 +0200
Subject: [PATCH] hwrng: stm32 - reduce RNG clock speed for STM32MP21
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Reducing clock speed in STM32MP21 improves the
stability

Change-Id: If511bc31bb85bce03c9cb102d0501122421802f0
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/501474
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Reviewed-by: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 drivers/char/hw_random/stm32-rng.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/char/hw_random/stm32-rng.c b/drivers/char/hw_random/stm32-rng.c
index f1f219de2dcb..fddebf164a79 100644
--- a/drivers/char/hw_random/stm32-rng.c
+++ b/drivers/char/hw_random/stm32-rng.c
@@ -568,7 +568,7 @@ static const struct stm32_rng_data stm32mp25_rng_data = {
 
 static const struct stm32_rng_data stm32mp21_rng_data = {
 	.has_cond_reset = true,
-	.max_clock_rate = 48000000,
+	.max_clock_rate = 8000000,
 	.nb_clock = 2,
 	.cr_config1_mask = GENMASK(27, 20),
 	.cr = 0x00800D00,
-- 
2.53.0

